tests/e2e/v2store_deprecation_test.go

```diff
@@ -182,6 +184,74 @@ func TestV2DeprecationSnapshotRecover(t *testing.T) {
 	assert.NoError(t, epc.Close())
 }
 
+func TestV2DeprecationSnapshotRecoverOldVersion(t *testing.T) {
+	e2e.BeforeTest(t)
+	dataDir := t.TempDir()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if !fileutil.Exist(e2e.BinPath.EtcdLastRelease) {
+		t.Skipf("%q does not exist", e2e.BinPath.EtcdLastRelease)
+	}
+	var snapshotCount uint64 = 10
+	epc := runEtcdAndCreateSnapshot(t, e2e.CurrentVersion, dataDir, snapshotCount)
+
+	lastVersion, err := e2e.GetVersionFromBinary(e2e.BinPath.EtcdLastRelease)
+	lastVersionStr := lastVersion.String()
+	lastClusterVersion := semver.New(lastVersionStr)
+	lastClusterVersion.Patch = 0
+	lastClusterVersionStr := lastClusterVersion.String()
+	t.Logf("etcdctl downgrade enable %s", lastVersionStr)
+	downgradeEnable(t, epc, lastVersion)
```

Comment:
Can you refactor this code to separate function?

Reply:
updating.